Network/AMQP/Internal.hs

```diff
@@ -230,7 +231,7 @@ openConnection'' connOpts = withSocketsDo $ do
         selectedSASL <- selectSASLMechanism handle serverMechanisms
 
         -- C: start_ok
-        writeFrame handle $ start_ok selectedSASL
+        writeFrame handle $ start_ok selectedSASL (coName connOpts)
```

Comment:
I think connOpts has a big enough scope that you can use it directly in the code below, so no need to add it as a function argument here.